main/src/cgeo/geocaching/maps/mapsforge/v6/layers/TapHandlerLayer.java

```diff
@@ -50,7 +50,9 @@ public boolean onTap(final LatLong tapLatLong, final Point layerXY, final Point
     public boolean onLongPress(final LatLong tapLatLong, final Point layerXY, final Point tapXY) {
         longTabLatLong = tapLatLong;
         requestRedraw();
-        tapHandler.onLongPress(tapXY);
+        if (!tapHandler.onLongPress(tapXY)) {
+            longTabLatLong = null;
+        }
```

Comment:
Works fine in the emulator. Although I think there should be a `requestRedraw()` after setting `longTabLatLong` to null. Or, why not inverting the if condition and setting the `longTapLatLong` only after the `onLongPress` had returned true?

Reply:
I need to set `longTapLatLong` before calling `tapHandler.onLongPress()` as it depends on a non-null value for it, but moving the `requestRedraw()` after the `if` clause seems to be ok. Updated PR will follow.